.circleci/config/jobs/website-docker-image.yml

```diff
@@ -7,26 +7,20 @@ steps:
   - run:
       name: Build Docker Image if Necessary
       command: |
-        # BUILD_FROM_REPO should ALWAYS be the main OSS Vault repo URL.
-        BUILD_FROM_REPO=git@github.com:hashicorp/vault.git
-        [ "$CIRCLE_REPOSITORY_URL" = "$BUILD_FROM_REPO" ] || {
-          echo "Not building website docker image for repo '$CIRCLE_REPOSITORY_URL' - we only build it for $BUILD_FROM_REPO."
-          exit 0
-        }
-        [ "$CIRCLE_BRANCH" = "master" ] || {
-          echo "Not building website docker image for branch '$CIRCLE_BRANCH' - we only build it for master."
-          exit 0
-        }
-        PACKAGE_LOCK_CHANGED="$( \
-          git diff --name-only $(git log --pretty=format:'%h' -n1 HEAD~1)...HEAD | \
-          grep -c website/package-lock.json; \
-        )" || true
-        [ $PACKAGE_LOCK_CHANGED -gt 0 ] || {
-          echo "Not building a new website docker image - dependencies have not changed."
-          exit 0
-        }
-        cd website/
-        docker build -t hashicorp/vault-website:$CIRCLE_SHA1 .
-        docker tag hashicorp/vault-website:$CIRCLE_SHA1 hashicorp/vault-website:latest
-        docker login -u $DOCKER_USER -p $DOCKER_PASS
-        docker push hashicorp/vault-website
+        # There is an edge case that would cause an issue here - if dependencies are updated to an exact copy
+        # of a previous version, for example if packge-lock.json is reverted, we need to manually push the new
+        # image to the "latest" tag
+        # Ignore job if running an enterprise build
+        IMAGE_TAG=$(cat website/Dockerfile website/package-lock.json | sha256sum | awk '{print $1;}')
```

Comment:
This would also have the benefit that you can lookup the git commit from the tag later, which you would lose by using the SHA256 sum of the file contents.

Reply:
In fact, maybe it would be better to rebuild the website on changes to any file in the `website/` directory? It's not clear that updates to content files would be reflected in the `package-lock.json` Using `git rev-parse` you could just pass the whole website directory instead, this is very cheap in git, as whole directories are tracked as trees. I would ultimately change it to:

```
IMAGE_TAG="$(git rev-list -n1 HEAD -- website/)"
```

Reply:
Amazing suggestions! Thank you for the review Sam.

On the whole website suggestion, we only actually need to rebuild the image if either of these paths have changed though, as the docker image just tracks dependencies, and not any changes to website files. Having it rebuild on any change to the website directory would cause a lot of useless rebuilds and would make local development much slower as anyone working on the website would have to download a new image pretty much every time they fire it up. I think for that reason I may stick with the first suggestion, unless there's something I'm missing here?

Reply:
Ah, I see... So it loads the content directly from master when the website is running? Cool stuff, nope I think I just didn't quite understand how it worked until now, thanks for explaining :)

Reply:
Using rev-list for the json file is an excellent idea and nicer for tracking dependencies than sha!  I'll adopt this pattern in future.  TIL!